packages/kit/src/runtime/client/client.js

```diff
@@ -0,0 +1,1233 @@
+import { onMount, tick } from 'svelte';
+import { writable } from 'svelte/store';
+import { coalesce_to_error } from '../../utils/error.js';
+import { normalize } from '../load.js';
+import { normalize_path } from '../../utils/url';
+import {
+	create_updated_store,
+	find_anchor,
+	get_base_uri,
+	get_href,
+	initial_fetch,
+	notifiable_store,
+	scroll_state
+} from './utils';
+
+import Root from '__GENERATED__/root.svelte';
+import { routes, fallback } from '__GENERATED__/manifest.js';
+
+const SCROLL_KEY = 'sveltekit:scroll';
+const INDEX_KEY = 'sveltekit:index';
+
+// We track the scroll position associated with each history entry in sessionStorage,
+// rather than on history.state itself, because when navigation is driven by
+// popstate it's too late to update the scroll position associated with the
+// state we're navigating from
+
+/** @typedef {{ x: number, y: number }} ScrollPosition */
+/** @type {Record<number, ScrollPosition>} */
+let scroll_positions = {};
+try {
+	scroll_positions = JSON.parse(sessionStorage[SCROLL_KEY]);
+} catch {
+	// do nothing
+}
+
+/** @param {number} index */
+function update_scroll_positions(index) {
+	scroll_positions[index] = scroll_state();
+}
+
+/**
+ * @param {{
+ *   target: Element;
+ *   session: App.Session;
+ *   base: string;
+ *   trailing_slash: import('types').TrailingSlash;
+ * }} opts
+ * @returns {import('./types').Client}
+ */
+export function create_client({ target, session, base, trailing_slash }) {
+	/** @type {Map<string, import('./types').NavigationResult>} */
+	const cache = new Map();
+
+	/** @type {Set<string>} */
+	const invalid = new Set();
+
+	const stores = {
+		url: notifiable_store({}),
+		page: notifiable_store({}),
+		navigating: writable(/** @type {import('types').Navigation | null} */ (null)),
+		session: writable(session),
+		updated: create_updated_store()
+	};
+
+	/** @type {{id: string | null, promise: Promise<import('./types').NavigationResult | undefined> | null}} */
+	const loading = {
```

Comment:
a better name for this might be `load_cache`

a related note - I just saw someone on Discord raise the issue that if they move their mouse between two links they will be prefetched over and over again. possibly we could make this more than a single-membered cache in the future

Reply:
renamed. re multiple prefetches, the current behaviour is by design, but if they want to open an issue then we can discuss further